Sources/MessagingInApp/Config/MessagingInAppConfigBuilder.swift

```diff
@@ -0,0 +1,65 @@
+import CioInternalCommon
+import Foundation
+
+/// Builder class designed to facilitate the creation of `MessagingInAppConfigOptions`.
+/// The builder pattern allows for a fluent and intuitive way to set up configuration options for
+/// the module, ensuring all required and optional settings are appropriately configured before the
+/// module is initialized.
+///
+/// **Usage Example:**
+///
+/// ```
+/// let config = MessagingInAppConfigBuilder(siteId, region: .US)
+///   // additional configuration as needed...
+///   .build()
+/// // Use `config` for initializing in-app module...
+/// ```
+public class MessagingInAppConfigBuilder {
+    // configuration options for MessagingInAppConfigOptions
+    private let siteId: String
+    private let region: Region
+
+    /// Initializes new `MessagingInAppConfigBuilder` with required configuration options.
+    /// - Parameters:
+    ///   - siteId: Workspace Site ID
+    ///   - region: Workspace Region
+    public init(siteId: String, region: Region) {
+        self.siteId = siteId
+        self.region = region
+    }
+
+    /// Builds and returns `MessagingInAppConfigOptions` instance from the configured properties.
+    public func build() -> MessagingInAppConfigOptions {
+        MessagingInAppConfigOptions(
+            siteId: siteId,
+            region: region
+        )
+    }
+}
+
+/// Defines errors that can occur during the `MessagingInAppConfigBuilder` building process.
+public enum MessagingInAppConfigBuilderError: Error {
+    case missingSiteId
+    case missingRegion
+}
+
+public extension MessagingInAppConfigBuilder {
+    /// Constants used to map each of the options in MessagingInAppConfigOptions.
+    enum Keys: String {
+        case siteId
+        case region
+    }
+
+    /// Constructs `MessagingInAppConfigOptions` by parsing and applying configurations from provided dictionary.
+    @available(iOSApplicationExtension, unavailable)
+    static func build(from dictionary: [String: Any]) throws -> MessagingInAppConfigOptions {
+        guard let siteId = dictionary[Keys.siteId.rawValue] as? String else {
+            throw MessagingInAppConfigBuilderError.missingSiteId
+        }
+        guard let region = (dictionary[Keys.region.rawValue] as? String).map(Region.getRegion) else {
+            throw MessagingInAppConfigBuilderError.missingRegion
+        }
+
+        return MessagingInAppConfigBuilder(siteId: siteId, region: region).build()
+    }
+}
```

Comment:
I think that this is for SDK wrappers, correct? 

Assuming that's true, I would like to see if there is a more maintainable, less boilerplate solution to building a config object from a dictionary. 

A couple ideas I can think of now would be to generate the code like we generate mocks and digraph today. Or, [use swift property wrappers](https://www.avanderlee.com/swift/property-wrappers/). 

Or, maybe we have a different solution altogether? Should we wait to implement this code when we build the RN SDK?

Reply:
Yes, you're right, this is for SDK wrappers.

I agree on preferring a more maintainable with less boilerplate solution for building a config object from a dictionary. However, that's outside the scope of this ticket. I would avoid removing this functionality since our native SDKs support it today, and its removal can become blocker for wrappers. So I plan to keep it for the time being. Given that it's utilized internally by our SDKs, we can modify the implementation later without impacting any customer-facing APIs.

Also, can you please create a ticket (or proposal) for this and add your idea (and any additional ideas if you have) to it? So we can consider in near future. May be when we start working on wrapper SDKs? 🤔